src/modm/driver/display/st7586s_impl.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021, Tomasz Wasilczyk
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+#ifndef MODM_ST7586S_HPP
+#error "Do not include this file directly. Include st7586s.hpp instead."
+#endif
+
+#include <modm/architecture/interface/assert.hpp>
+#include <modm/architecture/interface/gpio.hpp>
+
+namespace modm
+{
+
+template <typename SPI, typename CS, typename RST, typename DC, int Width, int Height>
+void
+St7586s<SPI, CS, RST, DC, Width, Height>::sendCommand(Command cmd, const void *data, size_t len)
+{
+	CS::reset();
+	DC::reset();  // command mode
+	SPI::transferBlocking(static_cast<uint8_t>(cmd));
+	DC::set();  // data mode
+	if (len > 0) {
+		SPI::transferBlocking(reinterpret_cast<const uint8_t*>(data), nullptr, len);
+	}
+	CS::set();
+	// exit with data mode on
+}
+
+template <typename SPI, typename CS, typename RST, typename DC, int Width, int Height>
+void
+St7586s<SPI, CS, RST, DC, Width, Height>::initialize()
+{
+	namespace payload = impl::st7586s::payload;
+
+	// Configure GPIO
+	RST::reset();
+	RST::setOutput();
```

Comment:
RST::setOutput(false); // same result

Reply:
Hah. I was actually hoping it would work this way, but the API scared me off. As it turns out, it wasn't API, but a mere implementation with swapped argument naming:

gpio_D4.hpp:
```
void setOutput(bool status);
```

While the TLD interface (gpio.hpp) has it:
```
void setOutput(bool value);
```

@salkinium: is this a typo? I was digging what `status` means and I had an impression it's something about pull-ups or port config.

Reply:
It's a typo, it should reflect the architecture interface API.